cucumber/authenticators_gcp/features/authn_gcf_token_errors.feature

```diff
@@ -0,0 +1,97 @@
+        resource: !webservice
+    """
+    And I have host "test-app"
+    And I grant group "conjur/authn-gcp/apps" to host "test-app"
+
+  Scenario: Token with service-account-id claim that does not match annotation is denied
+    Given I remove all annotations from host "test-app"
+    And I set "authn-gcp/service-account-id" annotation with value: "unknown-service-account-id" to host "test-app"
+    And I obtain a valid GCF identity token
+    And I save my place in the log file
+    When I authenticate with authn-gcp using a valid GCF identity token
+    Then it is unauthorized
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00049E Resource restriction 'authn-gcp/service-account-id' does not match resource in JWT token
+    """
+
+  Scenario: Token with service-account-email claim that does not match annotation is denied
+    Given I remove all annotations from host "test-app"
+    And I set "authn-gcp/service-account-email" annotation with value: "unknown-service-account-email" to host "test-app"
+    And I obtain a valid GCF identity token
+    And I save my place in the log file
+    When I authenticate with authn-gcp using a valid GCF identity token
+    Then it is unauthorized
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00049E Resource restriction 'authn-gcp/service-account-email' does not match resource in JWT token
+    """
+
+  Scenario: Token with valid service-account-id claim and service-account-email claim that does not match annotation is denied
+    Given I remove all annotations from host "test-app"
+    And I set "authn-gcp/service-account-id" annotation to function host "test-app"
+    And I set "authn-gcp/service-account-email" annotation with value: "unknown-service-account-email" to host "test-app"
+    And I obtain a valid GCF identity token
+    And I save my place in the log file
+    When I authenticate with authn-gcp using a valid GCF identity token
+    Then it is unauthorized
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00049E Resource restriction 'authn-gcp/service-account-email' does not match resource in JWT token
+    """
+
+  Scenario: Token with valid service-account-email claim and service-account-id claim that does not match annotation is denied
+    And I set "authn-gcp/service-account-email" annotation to function host "test-app"
+    And I set "authn-gcp/service-account-id" annotation with value: "unknown-service-account-id" to host "test-app"
+    And I obtain a valid GCF identity token
+    And I save my place in the log file
+    When I authenticate with authn-gcp using a valid GCF identity token
+    Then it is unauthorized
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00049E Resource restriction 'authn-gcp/service-account-id' does not match resource in JWT token
+    """
+
+  Scenario: Token with project-id host annotation is denied
```

Comment:
what happen if i declare service-account-email annotation and project-id annotation ? it suppose to work, but since they are going on the same path we need to verify it will not failed on project-id not exist.

Reply:
@eladkug i think test 13 under [error handling ](https://ca-il-confluence.il.cyber-ark.com/display/rndp/Conjur+GCP+authenticator+2nd+phase+-+Test+plan#ConjurGCPauthenticator2ndphase-Testplan-Errorhandling)cover this use case, WDYT?

Reply:
Yes, but i want it on automation as well.